src/ServerDaemon.chpl

```diff
@@ -0,0 +1,510 @@
+module ServerDaemon {
+    use FileIO;
+    use IO;
+    use Security;
+    use ServerConfig;
+    use ServerErrors;
+    use Time only;
+    use ZMQ only;
+    use Memory;
+    use FileSystem;
+    use IO;
+    use Logging;
+    use Path;
+    use MultiTypeSymbolTable;
+    use MultiTypeSymEntry;
+    use MsgProcessing;
+    use GenSymIO;
+    use Reflection;
+    use SymArrayDmap;
+    use ServerErrorStrings;
+    use Message;
+    use CommandMap, ServerRegistration;
+    use Errors;
+
+    enum ServerDaemonType {DEFAULT,INTEGRATION,METRICS}
+
+    private config const logLevel = ServerConfig.logLevel;
+    const sdLogger = new Logger(logLevel);
+
+    private config const daemonType = ServerDaemonType.DEFAULT;
+    
+    /**
+     * The ArkoudaServerDaemon class defines the run and shutdown 
+     * functions all derived classes must override
+     */
+    class ArkoudaServerDaemon {
+        var st = new owned SymTab();
+        var shutdownDaemon = false;
+        var port: int;
+ 
+        proc run() throws {
+            throw new NotImplementedError("run() must be overridden",
+                                          getLineNumber(),
+                                          getRoutineName(),
+                                          getModuleName());
+        }
+        
+        proc shutdown(user: string) throws {
+            this.shutdownDaemon = true;
+        }
+        
+        /*
+         * Converts the incoming request JSON string into RequestMsg object.
+         */
+        proc extractRequest(request : string) : RequestMsg throws {
+            var rm = new RequestMsg();
+            deserialize(rm, request);
+            return rm;
+        }
+    }
+
+    /**
+     * The BaseServerDaemon class serves as the base Arkouda server
+     * daemon which is run within the arkouda_server driver
+     */
+    class BaseServerDaemon : ArkoudaServerDaemon {
+        var serverToken : string;
+        var arkDirectory : string;
+        var serverMessage : string;
+        var reqCount: int = 0;
+        var repCount: int = 0;
```

Comment:
this is super nitpicky but i don't love how similar these var names are. It's not a problem but seems easy to get mixed up

Reply:
Also just realized these names are already in the server so yeah there's no need to change this